### PR TITLE
support multiple separators

### LINKED ISSUE #7: Support multiple separators
Sometimes we want an alternative separator (e.g. no separator) between particular components - e.g. see https://github.com/COSIMA/access-om3/issues/190#issuecomment-2252030909 - how to implement this?

### PR DESCRIPTION
**Pull request by aekiss**
closes https://github.com/COSIMA/make_diag_table/issues/7

### CODE REVIEW

File: make_diag_table.py [hunk 1]
```diff
@@ -34,27 +34,41 @@
...
+    return indict['file_name'].
+    If indict['file_name'] is a list, its elements can be a mixture of
+    strings and dictionaries.
+    String elements will be separated by file_name_separator.
+    Dictionary elements must have one element, whose key is a string and whose
+    value is a list of strings, which will be concatenated with key preceding
+    (separating) each one, i.e. the key is a custom separator for the items in
+    the list.
     """
     fn = indict['file_name']
```
Comment (minghangli-uni) [inline make_diag_table.py]:
```suggestion
    def get_subitem(item):
        sitem = str(indict[item])
        return indict['file_name_substitutions'].get(sitem, sitem)
         
    fn = indict['file_name']             
```

File: make_diag_table.py [hunk 2]
```diff
@@ -34,27 +34,41 @@
...
             else:
-                indict['file_name_date_section'] = '_'.join(fnd)
-        fn = [str(indict[k]) for k in fn]
-# TODO: make substitutions specific to particular components of indict['file_name']?
-        fn = [indict['file_name_substitutions'].get(v, v) for v in fn]
-        if indict['file_name_omit_empty']:
-            fn = [v for v in fn if v != '']
-        return indict['file_name_separator'].join(fn)
+                sitem = str(indict[item])
+                subitem = indict['file_name_substitutions'].get(sitem, sitem)
```
Comment (minghangli-uni) [inline make_diag_table.py]:
```suggestion
                subitem = get_subitem(ditem)
```

File: make_diag_table.py [hunk 3]
```diff
@@ -34,27 +34,41 @@
...
-            if fnd[-1][0] == '%':
-                # omit _ since date specification already supplies leading _
-                indict['file_name_date_section'] = '_'.join(fnd[0:-1]) + fnd[-1]
+        elements = []
+        for item in fn:
+            if isinstance(item, dict):  # use dict key as separator (assume exactly one key-value pair)
+                for sep, ditems in item.items():
+                    for ditem in ditems:
+                        sitem = str(indict[ditem])
+                        subitem = indict['file_name_substitutions'].get(sitem, sitem)
```
Comment (minghangli-uni) [inline make_diag_table.py]:
```suggestion
                        subitem = get_subitem(ditem)
```

**Review by minghangli-uni — APPROVED** [review]
Thanks @aekiss ! LGTM! Only a comment related to `make_diag_table.py`. Feel free to merge it!

**Comment by aekiss** [comment]
Thanks @minghangli-uni, good suggestion

**Pull request merged**